src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int ORIGIN = 1;
+    private static final int BOUND = 10;
+    private static final Random random = new Random();
+
+    public static int getRandomInt() {
+        return random.nextInt(ORIGIN, BOUND);
+    }
+
+    public static Figure getRandomFigure() {
+        FigureSupplier figureSupplier = new FigureSupplier();
+        int index = random.nextInt(Name.values().length);
+        Name name = Name.values()[index];
+        Figure figure = null;
+        switch (name.name()) {
+            case "CIRCLE" :
+                figure = new Circle(getRandomInt(), null);
+                break;
+            case "ISOSCELES_TRAPESOID" :
+                figure = new IsoscelesTrapezoid(getRandomInt(), getRandomInt(), getRandomInt());
+                break;
+            case "RIGHT_TRIANGLE" :
+                figure = new RightTriangle(getRandomInt(), getRandomInt());
+                break;
+            case "RECTANGLE" :
+                figure = new RightTriangle(getRandomInt(), getRandomInt());
+                break;
+            case "SQUARE" :
+                figure = new Square(getRandomInt());
+                break;
+            default:
+                figure = null;
+        }
+        return figure;
```

Comment:
Constructors will change according to previous comments but:
1) use swtich/case for enum, no need to transform it into String
2) use return from cases to make code simpler
3) use default section to return last figure, if it's not previous ones that it's obviously square
```suggestion
        switch (name) {
            case CIRCLE :
                return new Circle(getRandomInt(), null);
            case ISOSCELES_TRAPESOID :
                return new IsoscelesTrapezoid(getRandomInt(), getRandomInt(), getRandomInt());
            case RIGHT_TRIANGLE :
                return new RightTriangle(getRandomInt(), getRandomInt());
            case RECTANGLE :
                return new RightTriangle(getRandomInt(), getRandomInt());
            case default :
                return new Square(getRandomInt());
```